libminifi/test/TestBase.h

```diff
@@ -164,6 +165,31 @@ class LogTestController {
     return found;
   }
 
+  std::optional<std::smatch> matchesRegex(const std::string &regex_str,
+                std::chrono::seconds timeout = std::chrono::seconds(3),
+                std::chrono::milliseconds sleep_interval = std::chrono::milliseconds(200)) {
+    if (regex_str.length() == 0) {
+      return std::nullopt;
+    }
+    auto start = std::chrono::system_clock::now();
+    bool found = false;
+    bool timed_out = false;
+    std::regex matcher_regex(regex_str);
+    std::smatch match;
+    do {
+      std::string str = log_output.str();
+      found = std::regex_search(str, match, matcher_regex);
+      auto now = std::chrono::system_clock::now();
+      timed_out = std::chrono::duration_cast<std::chrono::milliseconds>(now - start) > std::chrono::duration_cast<std::chrono::milliseconds>(timeout);
```

Comment:
I don't think these `duration_cast`s are needed; `std::chrono::duration`s constructed with different units can be compared.
```suggestion
      timed_out = (now - start > timeout);
```

Reply:
Updated in dc9f6aa438e986a8ae6d1bacb85ca4fc677cb175